test/start-kubernetes.sh

```diff
@@ -0,0 +1,240 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+
+TEST_DIRECTORY=${TEST_DIRECTORY:-$(dirname $(readlink -f $0))}
+TEST_CONFIG=${TEST_CONFIG:-${TEST_DIRECTORY}/test-config.sh}
+source ${TEST_CONFIG}
+DEPLOYMENT_SUFFIX=${DEPLOYMENT_SUFFIX}
+DEPLOYMENT_ID=${DEPLOYMENT_ID:-k8s-test${DEPLOYMENT_SUFFIX}-pmem}
+GOVM_YAML=${GOVM_YAML:-$(mktemp --suffix $DEPLOYMENT_ID.yml)}
+REPO_DIRECTORY=${REPO_DIRECTORY:-$(dirname $TEST_DIRECTORY)}
+CLUSTER=${CLUSTER:-clear-kvm}
+TEST_DIRECTORY=${TEST_DIRECTORY:-$(dirname $(readlink -f $0))}
+RESOURCES_DIRECTORY=${RESOURCES_DIRECTORY:-${REPO_DIRECTORY}/_work/resources}
+WORKING_DIRECTORY="${WORKING_DIRECTORY:-${REPO_DIRECTORY}/_work/${CLUSTER}}"
+NODES=( $DEPLOYMENT_ID-master
+	$DEPLOYMENT_ID-worker1
+	$DEPLOYMENT_ID-worker2
+	$DEPLOYMENT_ID-worker3)
+CLOUD="${CLOUD:-true}"
+FLAVOR="${FLAVOR:-medium}"
+SSH_KEY="${SSH_KEY:-${RESOURCES_DIRECTORY}/id_rsa}"
+SSH_PUBLIC_KEY="${SSH_KEY}.pub"
+IMAGE_TAG="${IMAGE_TAG:-canary}"
+EFI="${EFI:-true}"
+KVM_CPU_OPTS="${KVM_CPU_OPTS:-\
+		-m 2G,slots=${TEST_MEM_SLOTS:-2},maxmem=34G -smp 4 \
+		-machine pc,accel=kvm,nvdimm=on}"
+EXTRA_QEMU_OPTS="${EXTRA_QWEMU_OPTS:-\
+		-object memory-backend-file,id=mem1,share=${TEST_PMEM_SHARE:-on},\
+		mem-path=/data/nvdimm0,size=${TEST_PMEM_MEM_SIZE:-32768}M \
+		-device nvdimm,id=nvdimm1,memdev=mem1,label-size=${TEST_PMEM_LABEL_SIZE:-2097152} \
+		-machine pc,nvdimm}"
+CLOUD_USER=${CLOUD_USER:-clear}
+CLOUD_IMAGE="${CLOUD_IMAGE:-$(\
+		curl -s https://download.clearlinux.org/image/latest-images |\
+		awk '/cloud/ {gsub(".xz",""); print $0}')}"
+IMAGE_URL=${IMAGE_URL:-http://download.clearlinux.org/image/${CLOUD_IMAGE}.xz}
+SSH_TIMEOUT=60
+SSH_ARGS="-oIdentitiesOnly=yes -oStrictHostKeyChecking=no \
+	-oUserKnownHostsFile=/dev/null -oLogLevel=error \
+	-i ${SSH_KEY}"
+CREATE_REGISTRY=${CREATE_REGISTRY:-false}
+CHECK_SIGNED_FILES=${CHECK_SIGNED_FILES:-true}
+
+function error_handler(){
+	local line="${1}"
+	echo  "Running the ${BASH_COMMAND} on function ${FUNCNAME[1]} at line ${line}"
+	delete_vms
+}
+
+function download_image(){
+    pushd $RESOURCES_DIRECTORY &>/dev/null
+	if [ -e "$CLOUD_IMAGE" ]; then
+		echo "$CLOUD_IMAGE found, skipping download"
+	else
+		case $CLOUD_USER in
+			clear)
+				ca_url="https://cdn.download.clearlinux.org/"
+				ca_url+="releases/${IMAGE_URL//[^0-9]}/"
+				ca_url+="clear/ClearLinuxRoot.pem"
+				curl -O ${IMAGE_URL}
+				curl -O ${IMAGE_URL}-SHA512SUMS
+				curl -O ${IMAGE_URL}-SHA512SUMS.sig
+				curl -O ${ca_url}
+				if [ "$CHECK_SIGNED_FILES" = "true" ]; then
+					if openssl smime -verify \
+					-in "${CLOUD_IMAGE}.xz-SHA512SUMS.sig" \
+					-inform DER \
+					-content "${CLOUD_IMAGE}.xz-SHA512SUMS" \
+					-CAfile "ClearLinuxRoot.pem"; then
+						unxz "${CLOUD_IMAGE}.xz"
+					else
```

Comment:
As expected, the check fails for me on Debian with:
```
Verification failure
140480049168576:error:21075075:PKCS7 routines:PKCS7_verify:certificate verify error:../crypto/pkcs7/pk7_smime.c:285:Verify error:unsupported certificate purpose
test/start-stop.make:5: recipe for target 'start' failed
make: *** [start] Error 2
```

Anyone else would be stuck at that point. Because I happen to know the issue, I could proceed with `CHECK_SIGNED_FILES=false make start`.

In the `else` branch we need add some output that explains what the problem is with a link to the issue that I filed and then explain how to proceed.

Reply:
Hi I added a message explaining the problem and referring to the issue